.github/workflows/release.yaml

```
@@ -25,178 +26,130 @@ env:
         uses: dtolnay/rust-toolchain@stable
         with:
           targets: ${{ env.TARGET }}
 
-      - name: Cache target directory
+      - name: Populate target directory from cache
         uses: Leafwing-Studios/cargo-cache@v1
 
-      - name: Install dependencies
-        run: |
-          sudo apt-get update; sudo apt-get install libasound2-dev libudev-dev libwayland-dev libxkbcommon-dev
+      - name: Install cargo-binstall
+        uses: cargo-bins/cargo-binstall@v1.7.4
 
-      - name: Build binary
-        run: |
-          cargo build --profile="${{ env.PROFILE }}" --target="${{ env.TARGET }}" --no-default-features --features=bevy/wayland
+      - name: Install trunk
+        run: cargo binstall trunk --no-confirm
 
-      - name: Prepare package
-        run: |
-          mkdir -p "${{ env.OUT_DIR }}"
-          cp "target/${{ env.TARGET }}/${{ env.PROFILE }}/${{ env.BINARY }}" "${{ env.OUT_DIR }}"
-          cp -r assets "${{ env.OUT_DIR }}" || true # Ignore error if `assets` folder does not exist
+      - name: Run trunk
+        run: trunk build --release --dist '${{ env.OUT_DIR }}'
 
-      - name: Compress package
-        working-directory: ./build/${{ env.PACKAGE }}
-        run: |
-          zip --recurse-paths ../"${{ env.PACKAGE }}.zip" "${{ env.PACKAGE_NAME }}"
+      - name: Finish package
+        working-directory: tmp/package
+        run: zip --recurse-paths '${{ env.PACKAGE }}${{ env.PACKAGE_EXT }}' '${{ env.PACKAGE_NAME }}'
 
       - name: Upload package to artifacts
         uses: actions/upload-artifact@v4
         with:
-          path: build/${{ env.PACKAGE }}.zip
-          name: ${{ env.PLATFORM }}
+          path: tmp/package/${{ env.PACKAGE }}${{ env.PACKAGE_EXT }}
+          name: package-${{ env.PLATFORM }}
           retention-days: 1
 
       - name: Upload package to Github release
         if: ${{ env.UPLOAD_PACKAGES_TO_GITHUB_RELEASE == 'true' }}
         uses: svenstaro/upload-release-action@v2
         with:
           repo_token: ${{ secrets.GITHUB_TOKEN }}
-          file: build/${{ env.PACKAGE }}.zip
-          asset_name: ${{ env.PACKAGE }}.zip
+          file: tmp/package/${{ env.PACKAGE }}${{ env.PACKAGE_EXT }}
+          asset_name: ${{ env.PACKAGE }}${{ env.PACKAGE_EXT }}
           release_name: ${{ env.VERSION }}
           tag: ${{ env.VERSION }}
           overwrite: true
 
-  # Build for Windows.
-  build-for-windows:
-    runs-on: windows-latest
-    needs: get-version
+  # Build and package a release for non-web platforms.
+  build:
+    needs:
+      - get-version
     env:
-      TARGET: x86_64-pc-windows-msvc
-      PROFILE: release-native
-      PLATFORM: windows
       VERSION: ${{ needs.get-version.outputs.version }}
-
+    strategy:
+      matrix:
+        include:
+          - platform: linux
+            targets: x86_64-unknown-linux-gnu
+            profile: release-native
+            features: bevy/wayland
+            package_ext: .zip
```

I'm not sure I'm a fan of `ubuntu-latest` being the default, especially since WASM is a separate job and is the only other target that could use Linux.

```suggestion
          - platform: linux
            targets: x86_64-unknown-linux-gnu
            profile: release-native
            features: bevy/wayland
            package_ext: .zip
            runner: ubuntu-latest
```

---

I'm considering combining the web job with the matrix now that it's possible (since there's no longer an intermediate step to upload a binary). Android would also use `ubuntu-latest`, and I think iOS would use `macos-latest`.

Not opposed to making it explicit, though.